src/hooks/sounds.tsx

```diff
@@ -35,35 +37,56 @@ export interface ExposedDataWithPlay extends ExposedData {
 
 export interface CustomSoundsState {
   soundConfigs: SoundConfigMap;
+  isLoading: boolean;
 }
 
 export const initialValue: CustomSoundsState = {
   soundConfigs: {},
+  isLoading: false,
 };
 
 export const CustomSoundsContext = createContext<CustomSoundsState>(
   initialValue
 );
 
-export const CustomSoundsProvider: React.FC = ({ children }) => {
+export type CustomSoundsType = {
+  loadingComponent: React.ReactNode;
+  waitTillConfigLoaded: boolean;
+};
```

Comment:
We use `interface` for props, and name it with the component name + `Props` suffix.

We also want these props to be optional rather than required:

```suggestion
export interface CustomSoundsProviderProps = {
  waitTillConfigLoaded?: boolean;
  loadingComponent?: React.ReactNode;
};
```